### PR TITLE
Add shellhooks plugin

### PR DESCRIPTION
**Pull request by adamruzicka**

### CODE REVIEW

**Comment by adamruzicka** [comment]
If I understand what's going on, enabling the shellhooks plugin will fail until https://github.com/theforeman/foreman_webhooks/pull/25 is merged and released.

**Comment by adamruzicka** [comment]
> If I understand what's going on, enabling the shellhooks plugin will fail until [theforeman/foreman_webhooks#25](https://github.com/theforeman/foreman_webhooks/pull/25) is merged and released.

And as a followup, if shellhooks requires a feature on the foreman side which is created by `foreman_webhooks`, should enabling `smart_proxy_shellhooks` also enable `foreman_webhooks`?

File: manifests/plugin/shellhooks.pp [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# = Foreman Proxy Shelhooks plugin
```
Comment (ehelms) [inline manifests/plugin/shellhooks.pp:1]:
```suggestion
# = Foreman Proxy Shellhooks plugin
```

File: templates/plugin/shellhooks.yml.erb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+# always enable HTTPS only
```
Comment (ehelms) [inline templates/plugin/shellhooks.yml.erb:2]: If this is true, why allow users to pass the `listen_on` parameter and thus change this?

File: manifests/plugin/shellhooks.pp (hunk 1 above)
Comment (adamruzicka) [reply manifests/plugin/shellhooks.pp:1]: That's like the 7th place where I made that typo -.-

File: templates/plugin/shellhooks.yml.erb (hunk 2 above)
Comment (adamruzicka) [reply templates/plugin/shellhooks.yml.erb:2]: My interpretation was that it works over http but allows *anyone* to run things on your proxy, hence using https only is recommended. My thinking was that if people know what they're doing, then we shouldn't prevent them from changing it

CC @lzap

Comment (ekohl) [reply templates/plugin/shellhooks.yml.erb:2]: That is indeed what we have for every module: a sane default but nothing should be hardcoded.

Comment (ehelms) [reply templates/plugin/shellhooks.yml.erb:2]: Can we change this message then? This message implies an absolute.

Comment (adamruzicka) [reply templates/plugin/shellhooks.yml.erb:2]: Once we agree on the exact wording in https://github.com/theforeman/smart_proxy_shellhooks/pull/1 , then I'll pull the changes here.

**Comment by lzap** [comment]
The comment was fixed upstream.

File: templates/plugin/shellhooks.yml.erb (hunk 2 above)
Comment (adamruzicka) [reply templates/plugin/shellhooks.yml.erb]: Done in https://github.com/theforeman/puppet-foreman_proxy/pull/651/commits/4979a02a18e4060b98c9be6bf71f6d928c701fd6

**Review by ehelms — APPROVED** [review]

**Pull request merged**